### PR TITLE
Get basic docker image working with oracle server

### PR DESCRIPTION
**Pull request by Christewart**
This is part of #2626 

This PR introduces configuration for building a docker image for the oracle server. 

## Building a docker image

You can build the docker image by doing the following inside of `sbt`. [Here](https://sbt-native-packager.readthedocs.io/en/latest/formats/docker.html#tasks) are the list of tasks you can use to build a docker image, we are going to use `stage

> oracleServer/compile;oracleServer/docker:stage

This builds a docker image and places it inside of the directory `app/oracle-server/target/docker/stage`

If you navigate to that directory, you can build the image by doing 

`docker build app/oracle-server/target/docker/stage/`

This will give you an image which you can see by doing 

Screenshot from 2021-02-16 10-55-54

## Running docker image

From the example above, you can see my docker image hash is `95b3d5989078`. We need this for running the image

Now we can start the image to create a running container with the [custom docker configuration](https://github.com/bitcoin-s/bitcoin-s/blob/7ad7a9fbf53f9d0534b009a50345a481ce2a7926/app/oracle-server/src/universal/docker-application.conf) following command

```
docker run -p 9998:9998 95b3d5989078
```
This should give you a log that looks like the following

Screenshot from 2021-02-16 10-58-53

Once this is done, you should be able to send requests to the oracle server with `bitcoin-s-cli` 

> ./bitcoin-s-cli getpublickey
4a4d9bd3e5f66211024599a3c29a0f6a9a3c2b75a90b5e0860da7d8f13fbd33e

Here is a screenshot of the following interaction

Screenshot from 2021-02-16 11-01-20

## Resources 

[Docker overview](https://docs.docker.com/get-started/overview/)
[Sbt native packager docker config](https://sbt-native-packager.readthedocs.io/en/latest/formats/docker.html#settings)

### CODE REVIEW

File: app/oracle-server/src/main/resources/reference.conf [hunk 1]
```diff
@@ -1,8 +1,9 @@
 bitcoin-s {
     oracle {
         rpcport = 9998
+        rpcbind = "0.0.0.0"
```
Comment (Christewart) [inline app/oracle-server/src/main/resources/reference.conf]: I think this is potentially a security problem, we don't want to publish on all interfaces unless this is running inside a docker container. So we probably need some sort of docker specific setting?

Comment (benthecarman) [reply app/oracle-server/src/main/resources/reference.conf]: Yeah this is dangerous, we shouldn't do this

Comment (Christewart) [reply app/oracle-server/src/main/resources/reference.conf]: It needs to be done inside the container, however we don't want to default it outside of the container.

File: build.sbt [hunk 2]
```diff
@@ -281,10 +280,14 @@ lazy val appCommonsTest = project
 lazy val oracleServer = project
   .in(file("app/oracle-server"))
   .settings(CommonSettings.prodSettings: _*)
+  .settings(
+    Compile / unmanagedResourceDirectories += baseDirectory.value / "src" / "universal"
```
Comment (Christewart) [inline build.sbt:284]: This is where the custom configuration is stored and added to the resources packaged in the artifact.

**Comment by benthecarman** [comment]
Might be nice to add a docs page explaining how to launch from docker or add to the getting started page

File: app/oracle-server/src/main/resources/reference.conf (hunk 1 above)
Comment (Christewart) [reply app/oracle-server/src/main/resources/reference.conf]: Ok this should be resolved in 4357b6f

**Comment by Christewart** [comment]
> Might be nice to add a docs page explaining how to launch from docker or add to the getting started page

Ok added in de9d767

Maybe you could try to set it up from the docs, and report back with what is confusing/misleading etc.

**Review by benthecarman — APPROVED** [review]

**Pull request merged**